drivers/scheduler/k8s/k8s.go

```diff
@@ -6539,15 +6622,28 @@ func (k *K8s) snapshotAndVerify(size resource.Quantity, data, snapName, namespac
 			Cause: fmt.Sprintf("restored pod is not ready. Error: %v", err),
 		}
 	}
+	mountPath, _ := pureutils.GetAppDataDir(restoredPod.Namespace)
 	// Run a cat command from within the pod to verify the content of dirtydata
-	cmdArgs := []string{"exec", "-it", restoredPod.Name, "-n", namespace, "--", "bin/cat", "/mnt/volume1/aaaa.txt"}
-	command := exec.Command("kubectl", cmdArgs...)
-	fileContent, err := command.CombinedOutput()
-	if err != nil {
-		return fmt.Errorf("error checking content of restored PVC: %s. Output: %s", err, string(fileContent))
-	}
-	if !strings.Contains(string(fileContent), data) {
-		return fmt.Errorf("restored volume does NOT contain data from original volume: expected to contain '%s', got '%s'", data, string(fileContent))
+	if *restoredPVC.Spec.VolumeMode == corev1.PersistentVolumeBlock {
+		ddCmd := fmt.Sprintf("dd if=%s status=none bs=38 count=1 skip=0", mountPath)
```

Comment:
this code look almost same as code from line 6691 to 6710. can you please move common code here into separate method

Reply:
Done